### PR TITLE
Update chooseEngineClientAlias() to iterating through keyTypes. 

### LINKED ISSUE #30203: Update Liberty code to call Java API KeyManager.chooseEngineClientAlias() per each key type
**Describe the bug**  
In WSX509KeyManager class, Liberty is only trying the first keyType : keyType[0]

**Steps to Reproduce**  
The user updated Java from 17.0.9 to 17.0.11.  It is suspected that in the user's environment, the keyType array in 17.0.9 coincidentally started with a keyType matching the certificate, whereas this is not the case in 17.0.11.

**Expected behavior**  
Liberty locates and sends a SSL client certificate correctly to the SSL server during SSLHandshake

**Diagnostic information:**  
- OpenLiberty version: This issue is not related to the OpenLiberty version
 - Affected feature(s) `transportSecurity-1.0`, `ssl-1.0`
 - Java Version: Java 17 (Currently investigating) 
 - server.xml configuration This issue is not related to a certain configuration

For GOOD log, WSX509KeyManager.chooseEngineClientAlias() finds a matching alias for RSA.
```
[10/16/24, 12:24:38:899 CEST] 0000007f WSX509KeyMana >  chooseEngineClientAlias Entry
                                 RSA
                                 C=US, ST=California, L=San Francisco, O=ExampleCorp, OU=Engineering, CN=www.example.com, EMAILADDRESS=admin@example.com
                                 SSLEngine[engine]
[10/16/24, 12:24:38:899 CEST] 0000007f WSX509KeyMana >  chooseClientAlias Entry  
                                 RSA
                                 C=US, ST=California, L=San Francisco, O=ExampleCorp, OU=Engineering, CN=www.example.com, EMAILADDRESS=admin@example.com
[10/16/24, 12:24:38:899 CEST] 0000007f JSSEHelper    >  getOutboundConnectionInfo Entry 
[10/16/24, 12:24:38:899 CEST] 0000007f ThreadContext 3   getOutboundConnectionInfo
[10/16/24, 12:24:38:899 CEST] 0000007f JSSEHelper    <  getOutboundConnectionInfo Exit  
                                 null
[10/16/24, 12:24:38:899 CEST] 0000007f SystemErr     R   javax.net.ssl|DEBUG|F7|HttpClient-1-Worker-0|2024-10-16 12:24:38.899 CEST|SunX509KeyManagerImpl.java:397|matching alias: exampleAlias
```

For ERROR log, WSX509KeyManager.chooseEngineClientAlias() finds no matching alias for [EC, EdDSA, RSASSA-PSS, RSA, DSA] when passed all at once.

```
[10/16/24, 12:18:17:909 CEST] 0000007f WSX509KeyMana >  chooseEngineClientAlias Entry
                                 EC
                                 EdDSA
                                 RSASSA-PSS
                                 RSA
                                 DSA
                                 C=US, ST=California, L=San Francisco, O=ExampleCorp, OU=Engineering, CN=www.example.com, EMAILADDRESS=admin@example.com
                                 SSLEngine[engine]

[10/16/24, 12:18:17:909 CEST] 0000007f WSX509KeyMana >  chooseClientAlias Entry  
                                 EC
                                 C=US, ST=California, L=San Francisco, O=ExampleCorp, OU=Engineering, CN=www.example.com, EMAILADDRESS=admin@example.com
                                 SSLEngine[engine]

[10/16/24, 12:18:17:909 CEST] 0000007f JSSEHelper    >  getOutboundConnectionInfo Entry 
[10/16/24, 12:18:17:909 CEST] 0000007f ThreadContext 3   getOutboundConnectionInfo
[10/16/24, 12:18:17:909 CEST] 0000007f JSSEHelper    <  getOutboundConnectionInfo Exit  
                                 null
[10/16/24, 12:18:17:910 CEST] 0000007f WSX509KeyMana <  chooseClientAlias (from JSSE) Exit  
                                 null
[10/16/24, 12:18:17:910 CEST] 0000007f WSX509KeyMana <  chooseEngineClientAlias Exit 
[10/16/24, 12:18:17:910 CEST] 0000007f SystemErr     R   javax.net.ssl|ALL|F7|HttpClient-1-Worker-0|2024-10-16 12:18:17.910 CEST|X509Authentication.java:278|No X.509 cert selected for [EC, EdDSA, RSASSA-PSS, RSA, DSA]
```

**Additional context**

### PR DESCRIPTION
**Pull request by una-tapa**
- [X] I have considered the risk of behavior change or other zero migration impact (https://github.com/OpenLiberty/open-liberty/wiki/Behavior-Changes).
- [X] If this PR fixes an Issue, the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN" (verify `release bug` label if applicable: https://github.com/OpenLiberty/open-liberty/wiki/Open-Liberty-Conventions).
- [ ] If this PR resolves an external Known Issue (including APARS), the description includes "Fixes #FILLMEIN" or "Resolves #FILLMEIN".
###############################################################################################

Resolves #30203

### CODE REVIEW

**Comment by una-tapa** [comment]
#build

**Comment by una-tapa** [comment]
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=446b7495-8159-4328-8702-07a0a299995d) - :x: completed with errors/failures.)

**Comment by una-tapa** [comment]
The build failed with one error in the com.ibm.ws.ui.tool.explore_fat.gui.nd.rw.monitor FAT bucket. 
The root cause was  "java.net.SocketTimeoutException: Read timed out error" when connecting to the Consul server. 
This indicates that the SSL/TLS handshake was successful, and the socket connection was established correctly, but the data could not be read within the timeout period. 
Since the code change involves the SSL handshake process, this error is unrelated to the code change in this Pull Request.

File: dev/com.ibm.ws.ssl/src/com/ibm/ws/ssl/core/WSX509KeyManager.java [hunk 1]
```diff
@@ -343,18 +343,26 @@ public String chooseEngineClientAlias(String[] keyType, Principal[] issuers, SSL
...
         if (null != customKM && customKM instanceof X509ExtendedKeyManager) {
             if (TraceComponent.isAnyTracingEnabled() && tc.isDebugEnabled())
                 Tr.debug(tc, "chooseEngineClientAlias, using customKM -> " + customKM.getClass().getName());
-            rc = ((X509ExtendedKeyManager) customKM).chooseEngineClientAlias(keyType, issuers, engine);
-        } else {
-            rc = chooseClientAlias(keyType[0], issuers);
-        }
+            alias = ((X509ExtendedKeyManager) customKM).chooseEngineClientAlias(keyType, issuers, engine);
+        } 
+        //Despite the Javadoc indicating support for passing an array of keyTypes,
```
Comment (utle) [inline dev/com.ibm.ws.ssl/src/com/ibm/ws/ssl/core/WSX509KeyManager.java]: Looks like you are missing the else here?

**Review by utle — CHANGES_REQUESTED** [review]

File: dev/com.ibm.ws.ssl/src/com/ibm/ws/ssl/core/WSX509KeyManager.java (hunk 1 above)
Comment (una-tapa) [reply dev/com.ibm.ws.ssl/src/com/ibm/ws/ssl/core/WSX509KeyManager.java]: added else as suggested by Ut. Thanks!

**Comment by una-tapa** [comment]
!build ([view Open Liberty Personal Build](https://libh-proxy1.fyre.ibm.com/cognitive/pipelineAnalysis.html?pipelineId=98d8e5d5-f2d2-451b-81d4-95646c1cc9df) - :x: completed with errors/failures.)

_Note: Target locations of links might be accessible only to IBM employees._

**Review by utle — APPROVED** [review]

**Pull request merged**